include/dxc/Support/HLSLOptions.h

```diff
@@ -167,6 +168,13 @@ class DxcOpts {
   bool IsRootSignatureProfile();
   bool IsLibraryProfile();
 
+  // Helpers to clarify interpretation of flags for behavior in implementation
+  bool IsDebugInfoEnabled();    // Zi || Fd
+  bool EmbedDebugInfo();        // Zi && !Fd && !Qstrip_debug
+  bool EmbedPDBName();          // Zi || Fd || pdb_alt_path
+  bool DebugFileIsDirectory();  // Fd ends in '\\'
```

Comment:
(style) you could name this `bool AutoPDBName()` and provide a `std::string GetAutoPDBDirectory()` that asserts if `!AutoPDBName()`, just to hide a bit better the trailing-slash implementation.

Reply:
Sure, that could be nicer.